diff --git a/ffmpeg.c b/ffmpeg.c
index 9792281..a31ffd7 100644
--- a/ffmpeg.c
+++ b/ffmpeg.c
@@ -858,6 +858,12 @@ static void do_video_out(AVFormatContext *s,
 
     /* by default, we output a single frame */
     nb_frames = 1;
+    {
+      // long pts       = ist->decoded_frame->pts; // <== jan2014 no longer set
+      long pts          = av_rescale_q(ist->pts, AV_TIME_BASE_Q, ist->st->time_base);
+      s->secxxx         = (pts==AV_NOPTS_VALUE || pts<0 ? 0 : (long int)floor( av_q2d(ist->st->time_base) * pts ));
+      snprintf(s->debugxxx, 60, "pts:%ld tbn:%d/%d frame:%d ==> %ld\n", pts, ist->st->time_base.num, ist->st->time_base.den, ist->st->codec->frame_number, s->secxxx);
+    }
 
     format_video_sync = video_sync_method;
     if (format_video_sync == VSYNC_AUTO) {
diff --git a/libavformat/img2enc.c b/libavformat/img2enc.c
index 37dfbec..845fb7c 100644
--- a/libavformat/img2enc.c
+++ b/libavformat/img2enc.c
@@ -97,6 +97,10 @@ static int write_packet(AVFormatContext *s, AVPacket *pkt)
                    img->img_number, img->path);
             return AVERROR(EINVAL);
         }
+
+if (av_get_frame_filename(filename, sizeof(filename), img->path, s->secxxx) < 0) return AVERROR(EINVAL);
+av_log(NULL, AV_LOG_WARNING, s->debugxxx);
+
         for (i = 0; i < 4; i++) {
             if (avio_open2(&pb[i], filename, AVIO_FLAG_WRITE,
                            &s->interrupt_callback, NULL) < 0) {
diff --git a/libavformat/avformat.h b/libavformat/avformat.h
index f21a1d6..0fadcc5 100644
--- a/libavformat/avformat.h
+++ b/libavformat/avformat.h
@@ -1739,7 +1739,7 @@ typedef struct AVFormatContext {
      * - demuxing: Set by user.
      */
     uint8_t *dump_separator;
-} AVFormatContext;
+char debugxxx[60]; long secxxx; } AVFormatContext;
 
 int av_format_get_probe_score(const AVFormatContext *s);
 AVCodec * av_format_get_video_codec(const AVFormatContext *s);
