diff --git a/libavcodec/libtheoraenc.c b/libavcodec/libtheoraenc.c
index 30a4b28..311ecf3 100644
--- a/libavcodec/libtheoraenc.c
+++ b/libavcodec/libtheoraenc.c
@@ -205,7 +205,9 @@ static av_cold int encode_init(AVCodecContext* avc_context)
                 * an int value
          */
         t_info.quality        = av_clipf(avc_context->global_quality / (float)FF_QP2LAMBDA, 0, 10) * 6.3;
-        t_info.target_bitrate = 0;
+        t_info.target_bitrate = avc_context->bit_rate;
+        av_log(avc_context, AV_LOG_WARNING, "[tracey patched] t_info.target_bitrate => %d\n", t_info.target_bitrate);
+        av_log(avc_context, AV_LOG_WARNING, "[tracey patched] t_info.quality => %d/63\n", t_info.quality);
     } else {
         t_info.target_bitrate = avc_context->bit_rate;
         t_info.quality        = 0;
